sdk/src/main/scala/com/horizen/block/SidechainBlockHeader.scala

```diff
@@ -0,0 +1,152 @@
+import scorex.core.{NodeViewModifier, bytesToId, idToBytes}
+import scorex.crypto.hash.Blake2b256
+import scorex.util.ModifierId
+import scorex.util.serialization.{Reader, Writer}
+
+@JsonView(Array(classOf[Views.Default]))
+@JsonIgnoreProperties(Array("messageToSign", "serializer"))
+case class SidechainBlockHeader(
+                                 version: Block.Version,
+                                 @JsonSerialize(using = classOf[ScorexModifierIdSerializer]) parentId: ModifierId,
+                                 timestamp: Block.Timestamp,
+                                 forgerBox: ForgerBox,
+                                 @JsonSerialize(using = classOf[MerklePathSerializer]) forgerBoxMerklePath: MerklePath,
+                                 @JsonSerialize(using = classOf[VRFProofSerializer]) vrfProof: VRFProof,
+                                 sidechainTransactionsMerkleRootHash: Array[Byte], // don't need to care about MC2SCAggTxs here
+                                 mainchainMerkleRootHash: Array[Byte], // root hash of MainchainBlockReference.dataHash() root hash and MainchainHeaders root hash
+                                 ommersMerkleRootHash: Array[Byte], // build on top of Ommer.id()
+                                 ommersCumulativeScore: Long, // to be able to calculate the score of the block without having the full SB. For future
+                                 signature: Signature25519
+                               ) extends BytesSerializable {
+
+  override type M = SidechainBlockHeader
+
+  override def serializer: ScorexSerializer[SidechainBlockHeader] = SidechainBlockHeaderSerializer
+
+  @JsonSerialize(using = classOf[ScorexModifierIdSerializer])
+  lazy val id: ModifierId = bytesToId(Blake2b256(Bytes.concat(messageToSign, signature.bytes)))
+
+  lazy val messageToSign: Array[Byte] = {
+    Bytes.concat(
+      idToBytes(parentId),
+      Longs.toByteArray(timestamp),
+      forgerBox.id(),
+      vrfProof.bytes, // TO DO: is it ok or define vrfProof.id() ?
+      forgerBoxMerklePath.bytes(), // TO DO: is it ok?
+      sidechainTransactionsMerkleRootHash,
+      mainchainMerkleRootHash,
+      ommersMerkleRootHash,
+      Longs.toByteArray(ommersCumulativeScore)
+    )
+  }
+
+  def semanticValidity(): Boolean = {
+    if(parentId == null || parentId.length != 64
+      || sidechainTransactionsMerkleRootHash == null || sidechainTransactionsMerkleRootHash.length != 32
+      || mainchainMerkleRootHash == null || mainchainMerkleRootHash.length != 32
+      || ommersMerkleRootHash == null || ommersMerkleRootHash.length != 32 || ommersCumulativeScore < 0
+      || forgerBox == null || signature == null)
+      return false
+
+    // TODO: do we need to allow Blocks from nearest future as in MC for Ouroboros?
+    // Check if timestamp is valid and not too far in the future
+    if(timestamp <= 0 || timestamp > Instant.now.getEpochSecond + 2 * 60 * 60) // 2 * 60 * 60 like in Horizen MC
```

Comment:
1) check in Ouroboros Praos paper if future time should be checked
2) In case it is needed to be checked, move the verification after signature and provide a validation result that will not cause the block to be marked as invalid and also avoid banning the sender